### ISSUE TITLE
Integrate detr

### ISSUE THREAD

**Issue by lgvaz** [airctic/icevision#40]
## 🚀 Feature
[Detr](https://github.com/facebookresearch/detr) is an amazing new approach to object detection just launched by facebook. Pretained weights are available to hub, so naturally this issue is a bit related to #38.

**Describe the solution you'd like**
Let's divide this task into three separate parts:
- [x] Model inference
- [ ] Train from scratch: Should be easier to implement and is supported in the original code
- [ ] Fine tuning: Not officially tested in the original code, will be a bit harder to implement.

**Additional context**
No other library supports this yet, let's goooo!!  🚀 🚀 🚀

**Comment by lgvaz** [comment]
The main disadvantage of detr is that it's very slow to train (because it uses transformers) so training from scratch will be out of the picture for most people...

Maybe we should work on inference before training

**Comment by lgvaz** [comment]
The inference notebook provided [here](https://colab.research.google.com/github/facebookresearch/detr/blob/colab/notebooks/detr_demo.ipynb) can be a good start

It's very straight forward to get predictions, pass in a batch of images and we get back a list of logits for the classes and the relative position of the boxes
```python
outputs = model(img)
# keep only predictions with 0.7+ confidence
probas = outputs['pred_logits'].softmax(-1)[0, :, :-1]
keep = probas.max(-1).values > 0.7
# convert boxes from [0; 1] to image scales
bboxes_scaled = rescale_bboxes(outputs['pred_boxes'][0, keep], im.size)
``` 

It's important to normalize our input image:
```python
T.Normalize([0.485, 0.456, 0.406], [0.229, 0.224, 0.225])
```

**Comment by lgvaz** [comment]
We can start by copying the model provided in the colab notebook.

Then we need to convert the model predictions to actual classes.
* `str` for classes names?
* `BBox` for bounding boxes

Can we make the convention that a model `predict` always returns the standard classes?

**Comment by oke-aditya** [comment]
Well, I have started a bit on this. I need to test my pipeline and train the model. It supports fine-tuning.
I will write Dream API here.
I am unsure of `backbone=None`
```
model = MantisDetr(num_classes, num_queries, backbone=something, pretrained=True)
train_loader = model.train_loader()
val_loader = model.val_loader()
```
After this we have lightning / fastai engine with detr. We take care of training of loss, and the model.
The problem is Hungarian Matcher and the loss function which they use, we need to take the code from official library.
Either copy paste and maintain it. Or we need to find some way to connect it.
But this API is simple, consistent with FasterRCNN. Supports only backbones which detr supports. Not all for now, but we can extend later.

**Comment by oke-aditya** [comment]
But this will take detr out of the hub and make it simple to integrate, provide backbones, provide some extensibility.
It allows to edit most parameters, which user needs. He would rarely touch detr training and loss fn.
